gulpfile.babel.js

```diff
@@ -150,7 +185,29 @@ const compileCss = () =>
 
 export const generateCss = gulp.series(cleanCss, compileCss);
 
-export const build = gulp.parallel(validateSass, generateCss);
+export const lintJs = () => {
+  return gulp.src([config.files.src.js, `!${config.dir.src.js}/dist/**/*.js`])
+    .pipe(eslint())
+    .pipe(eslint.format())
+    .pipe(eslint.failAfterError())
+};
+
+const transpileAndBundleJs = (done) => {
+  // Adapted from https://stackoverflow.com/questions/33558396/gulp-webpack-or-just-webpack
+  webpack(webpackConfig).run(handleJsBuild(done));
+};
+
+const testJs = () => {
+  return gulp.src(config.dir.test.js)
+    // TODO: remove passWithNoTests once js work has started
+    .pipe(jest({ 'passWithNoTests': true}));
+};
+
+export const buildCss = gulp.series(validateSass, generateCss);
+
+export const buildJs = gulp.series(lintJs, transpileAndBundleJs, testJs);
+
+export const build = gulp.parallel(buildCss, buildJs);
```

Comment:
Triggering an error in `lintJs` sees

```
[11:01:01] 'build' errored after 442 ms
[11:01:01] The following tasks did not complete: <series>, <parallel>, lintSass, testSass
[11:01:01] Did you forget to signal async completion?
```

Though not sure why.

Reply:
Changing `lintSass`, `testSass` and `compileCss` to calling `done` rather than returning resolves it. But shouldn't be needed?

Reply:
Weird, no that shouldn't be necessary.

Looks like it's the `lintJs` task's `.pipe(eslint.failAfterError())` line that is causing the problem, but without it the task doesn't fail on a lint error.  At some point am planning to try running ESLint from Webpack rather than Gulp anyway, which might get rid of this irritation. As it's not a big problem, I think we can safely ignore it for now.